draft-wussler-openpgp-pqc.md

```diff
@@ -1405,31 +1474,51 @@ computer attacks.
 
 # Security Considerations
 
+## Hashing in ECC-KEM
+
+Our construction of the ECC-KEMs, in particular the final hashing step in
+encapsulation and decapsulation that produces the `eccKeyShare`, is standard
+and known as hashed ElGamal key encapsulation, a hashed variant of ElGamal
+encryption. It ensures IND-CCA2 security in the random oracle model under some
+Diffie-Hellman intractability assumptions [CS03].
+
 ## Key combiner {#sec-key-combiner}
 
 For the key combination in {{kem-key-combiner}} this specification limits
 itself to the use of KMAC. The sponge construction used by KMAC was proven to
-be indifferentiable from a random oracle {{SPONGE}}. This means, that in
+be indifferentiable from a random oracle {{BDPA08}}. This means, that in
 contrast to SHA2, which uses a Merkle-Damgard construction, no HMAC-based
 construction is required for key combination. Except for a domain separation it
 is sufficient to simply process the concatenation of any number of key shares
 when using a sponge-based construction like KMAC. The construction using KMAC
 ensures a standardized domain separation. In this case, the processed message
 is then the concatenation of any number of key shares.
 
-More precisely, for a given capacity `c` the indifferentiability proof shows that
-assuming there are no weaknesses found in the Keccak permutation, an attacker
-has to make an expected number of `2^(c/2)` calls to the permutation to tell
-KMAC from a random oracle. For a random oracle, a difference in only a single
-bit gives an unrelated, uniformly random output. Hence, to be able to
-distinguish a key `K`, derived from shared keys `K1` and `K2` as
+More precisely, for a given capacity `c` the indifferentiability proof shows
+that assuming there are no weaknesses found in the Keccak permutation, an
+attacker has to make an expected number of `2^(c/2)` calls to the permutation
+to tell KMAC from a random oracle. For a random oracle, a difference in only a
+single bit gives an unrelated, uniformly random output. Hence, to be able to
+distinguish a key `K`, derived from shared keys `K1` and `K2` (and ciphertexts
+`C1` and `C2`) as
 
-    K = KMAC(domainSeparation, counter || K1 || K2 || fixedInfo,
+    K = KMAC(domainSeparation, counter || K1 || C1 || K2 || C2 || fixedInfo,
```

Comment:
Isn't this inconsistent with the hashing done above?
I would really prefer this version, as we can simply argue it's secure and more performant

Reply:
Sure, but then it should be expressed as 
```
K = KMAC(domainSeparation, counter || H( K1 || C1 ) || H( K2 || C2 ) || fixedInfo,
```
In this consideration we don't account for the additional hashing step

Reply:
This is possible, but not necessary (from a security point of view).

Reply:
If you use that version of the combiner, but indeed, then you have an unnecessary hashing for Kyber. Moreover, do you have a KEM description which defines an insecure KEM (which should at least be highlighted in that section in this case). The danger that I see is that extensions to new algorithms may render the whole thing insecure if we are not explicit where we do require what... (and in the previous version of the draft this was not clear at all)

Reply:
So basically what this construction wants is the following?
```K = KMAC(domainSeparation, counter || H( SS1 || C1 ) || C1 || H( SS2 || C2 ) || C2 || fixedInfo,```
Since K1 is already defined here as H( SS1 || C1 )?

Even if it's not necessary from the security standpoint, I'd like to keep this sentence as close as possible to what we're doing in order to cause less confusion